java/vector/src/main/java/org/apache/arrow/vector/ExtensionTypeVector.java

```diff
@@ -0,0 +1,231 @@
+
+import io.netty.buffer.ArrowBuf;
+
+/**
+ * A vector that wraps an underlying vector, used to help implement extension types.
+ * @param <T> The wrapped vector type.
+ */
+public abstract class ExtensionTypeVector<T extends BaseValueVector & FieldVector> extends BaseValueVector implements
+    FieldVector {
+
+  private final T underlyingVector;
+
+  public ExtensionTypeVector(String name, BufferAllocator allocator, T underlyingVector) {
+    super(name, allocator);
+    this.underlyingVector = underlyingVector;
+  }
+
+  /** Get the underlying vector. */
+  public T getUnderlyingVector() {
+    return underlyingVector;
+  }
+
+  @Override
+  public void allocateNew() throws OutOfMemoryException {
+    this.underlyingVector.allocateNew();
+  }
+
+  @Override
+  public boolean allocateNewSafe() {
+    return this.underlyingVector.allocateNewSafe();
+  }
+
+  @Override
+  public void reAlloc() {
+    this.underlyingVector.reAlloc();
+  }
+
+  @Override
+  public void setInitialCapacity(int numRecords) {
+    this.underlyingVector.setInitialCapacity(numRecords);
+  }
+
+  @Override
+  public int getValueCapacity() {
+    return this.underlyingVector.getValueCapacity();
+  }
+
+  @Override
+  public void reset() {
+    this.underlyingVector.reset();
+  }
+
+  @Override
+  public Field getField() {
+    return this.underlyingVector.getField();
+  }
+
+  @Override
+  public MinorType getMinorType() {
+    return MinorType.EXTENSIONTYPE;
+  }
+
+  @Override
+  public TransferPair getTransferPair(String ref, BufferAllocator allocator) {
+    return underlyingVector.getTransferPair(ref, allocator);
+  }
+
+  @Override
+  public TransferPair getTransferPair(String ref, BufferAllocator allocator, CallBack callBack) {
+    return underlyingVector.getTransferPair(ref, allocator, callBack);
+  }
+
+  @Override
+  public TransferPair makeTransferPair(ValueVector target) {
+    return underlyingVector.makeTransferPair(target);
+  }
+
+  @Override
+  public FieldReader getReader() {
+    return underlyingVector.getReader();
+  }
+
+  @Override
+  public int getBufferSize() {
+    return underlyingVector.getBufferSize();
+  }
+
+  @Override
+  public int getBufferSizeFor(int valueCount) {
+    return underlyingVector.getBufferSizeFor(valueCount);
+  }
+
+  @Override
+  public ArrowBuf[] getBuffers(boolean clear) {
+    return underlyingVector.getBuffers(clear);
+  }
+
+  @Override
+  public ArrowBuf getValidityBuffer() {
+    return underlyingVector.getValidityBuffer();
+  }
+
+  @Override
+  public ArrowBuf getDataBuffer() {
+    return underlyingVector.getDataBuffer();
+  }
+
+  @Override
+  public ArrowBuf getOffsetBuffer() {
+    return underlyingVector.getOffsetBuffer();
+  }
+
+  @Override
+  public int getValueCount() {
+    return underlyingVector.getValueCount();
+  }
+
+  @Override
+  public void setValueCount(int valueCount) {
+    underlyingVector.setValueCount(valueCount);
+  }
+
+  /** Get the object at the specified index. */
```

Comment:
Might be worth mentioned something like Gets the "extension object" or something else to make it more obvious why this method doesn't delegate like the rest of them.

Reply:
Makes sense, I've expanded the docstring.